packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -147,7 +146,25 @@ export default function EntitiesSavedStates( { close, onSave = identity } ) {
 		}
 	};
 
-	const saveCheckedEntitiesAndActivate = () => {
+	const {
+		additionalPrompt,
+		isDirty: isDirtyContext,
+		onSave,
+		saveEnabled: saveEnabledContext,
+		saveLabel,
+		setIsDirty: setIsDirtyContext,
+	} = useContext( EntitiesSavedStatesContext );
+	const isDirty =
+		isDirtyContext ??
+		dirtyEntityRecords.length - unselectedEntities.length > 0;
+	useEffect( () => {
+		setIsDirtyContext(
+			dirtyEntityRecords.length - unselectedEntities.length > 0
+		);
```

Comment:
Instead of an effect here can we just set the initial value to be this? Generally setState within an effect is a clue that we should be doing this calculation on render.

https://react.dev/learn/you-might-not-need-an-effect

Reply:
@getdave CC: @scruffian 
Thank you for your comment!
I lifted states up and added how I did it in the PR description.
I'll add Testing Instructions/Screenshots and look into the failing tests tomorrow if the implementation makes sense.